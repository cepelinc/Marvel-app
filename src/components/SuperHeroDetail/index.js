import React from 'react';
import PropTypes from 'prop-types';
import style from './style.styled';
const Container = style.container;
const Title = style.title;
const DetailContainer = style.detailContainer;
const Item = style.item;


const SuperHeroDetail = ({name, thumbnail, description, urls}) => (
  <Container>
    <Title>{name}</Title>
    <DetailContainer>
      <Item><img src={`${thumbnail.path}.${thumbnail.extension}`}/></Item>
      <Item><h3>Biography</h3> {description || 'no biography available'}</Item>
      <Item>
        <h3>SuperLinks</h3>
        {urls.length ? <ul>{urls.map(link => <li key={`${link.type}`}><a href={link.url} target="_blank" rel="noopener noreferrer" >{link.type}</a></li>)}</ul> : 'no links available'}
      </Item>
    </DetailContainer>
  </Container>
)

SuperHeroDetail.propTypes = {
  name: PropTypes.string,
  thumbnail: PropTypes.object,
  description: PropTypes.string,
  urls: PropTypes.array
};

export default SuperHeroDetail
