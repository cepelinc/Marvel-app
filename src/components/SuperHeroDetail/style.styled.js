
import styled from 'styled-components';

export default {
  container: styled.div`
    flex: 1;
    margin: 2%;
    box-shadow: 0px 0px 10px 10px rgba(196,196,196,1);
    overflow-y: auto;
  `,
  title: styled.h1`
    background: blue;
    margin: 0;
    padding: 20px;
    color: white;
    text-align: center;
  `,
  detailContainer: styled.div`
    display: flex;
    justify-content: space-around;

  `,
  item: styled.div`
    flex: .33;
    background: #ECECEC;
    padding: 20px;
    margin: 10px;
    height: 78vh;

    img{
      width: 100%
    }
  `
}
