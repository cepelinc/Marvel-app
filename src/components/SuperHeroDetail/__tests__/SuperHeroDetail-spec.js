import React from 'react'
import renderer from 'react-test-renderer';
import SuperHeroDetail from '@components/SuperHeroDetail'

describe('HeroesList', () => {
  const mockData = {
    mame: "Spiderman",
    thumbnail: {
      path: 'path',
      extension: 'jpg'
    },
    description: 'description',
    urls: [],
  };
  it('matches snapshot', () => {
    const Component = renderer.create(
      <SuperHeroDetail {...mockData} />)
      .toJSON()

    expect(Component).toMatchSnapshot()
  })
})
