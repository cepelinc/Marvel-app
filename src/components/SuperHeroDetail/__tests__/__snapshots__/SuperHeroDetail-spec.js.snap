// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesList matches snapshot 1`] = `
<div
  className="stylestyled__container-sc-1yxe6pz-0 gLyuMP"
>
  <h1
    className="stylestyled__title-sc-1yxe6pz-1 dOgraA"
  />
  <div
    className="stylestyled__detailContainer-sc-1yxe6pz-2 cZeaIC"
  >
    <div
      className="stylestyled__item-sc-1yxe6pz-3 bRHJTS"
    >
      <img
        src="path.jpg"
      />
    </div>
    <div
      className="stylestyled__item-sc-1yxe6pz-3 bRHJTS"
    >
      <h3>
        Biography
      </h3>
       
      description
    </div>
    <div
      className="stylestyled__item-sc-1yxe6pz-3 bRHJTS"
    >
      <h3>
        SuperLinks
      </h3>
      no links available
    </div>
  </div>
</div>
`;
